homeassistant/components/here_travel_time/sensor.py

```diff
@@ -91,32 +94,43 @@
 
 NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
 
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Inclusive(
+            CONF_DESTINATION_LATITUDE, "destination_coordinates"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_DESTINATION_LONGITUDE, "destination_coordinates"
+        ): cv.longitude,
+        vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
+        vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
+        vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
+        vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
+        vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
+        vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
+        vol.Exclusive(CONF_ARRIVAL, "arrival_departure"): cv.time,
+        vol.Exclusive(CONF_DEPARTURE, "arrival_departure"): cv.time,
+        vol.Optional(CONF_DEPARTURE, default="now"): cv.time,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+        vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(ROUTE_MODE),
+        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+    }
+)
+
 PLATFORM_SCHEMA = vol.All(
     cv.has_at_least_one_key(CONF_DESTINATION_LATITUDE, CONF_DESTINATION_ENTITY_ID),
     cv.has_at_least_one_key(CONF_ORIGIN_LATITUDE, CONF_ORIGIN_ENTITY_ID),
-    PLATFORM_SCHEMA.extend(
-        {
-            vol.Required(CONF_API_KEY): cv.string,
-            vol.Inclusive(
-                CONF_DESTINATION_LATITUDE, "destination_coordinates"
-            ): cv.latitude,
-            vol.Inclusive(
-                CONF_DESTINATION_LONGITUDE, "destination_coordinates"
-            ): cv.longitude,
-            vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
-            vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
-            vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
-            vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
-            vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
-            vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
-            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-            vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
-            vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(
-                ROUTE_MODE
-            ),
-            vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
-            vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
-        }
+    vol.Any(
```

Comment:
This will generate confusing error messages. Use `cv.key_value_schemas` for this.

Reply:
I tried for some time now but I cannot get it working.
The code here says that IF `CONF_MODE` is `TRAVEL_MODE_PUBLIC_TIME_TABLE` THEN `CONF_ARRIVAL` is required.

How do I that with the new validator?

Reply:
scratch that. I got it